### PR TITLE
AWS v12.1.4-2020091600 release for Helm 3

### PR DESCRIPTION
**Pull request by tomahawk28**
Same as https://github.com/giantswarm/releases/pull/432 but for AWS installations.

### CODE REVIEW

**Comment by tomahawk28** [comment]
The new AWS release failed due to flatcar upgrade constraints. 
```
go: downloading github.com/globalsign/mgo v0.0.0-20181015135952-eeefdecb41b8
--- FAIL: Test_Releases (0.21s)
    --- FAIL: Test_Releases/case_1:_aws_releases_are_valid (0.17s)
        releases_test.go:376: Release v12.2.0 does not meet the requested version requirements:
            requested: flatcar: >= 2512.4.0 	actual: 
```
@stone-z @giantswarm/team-batman is trying to testing AWS release which includes helm 3. It's not mean to contains any other system changes but for app changes. Can we simply update the constraint to start from 12.2.1 platform release version?

**Comment by alex-dabija** [comment]
Team Firecracker is already working on a v12.2.0 release: https://github.com/giantswarm/releases/pull/431 .

**Comment by stone-z** [comment]
> Can we simply update the constraint to start from 12.2.1 platform release version?

It is confusing to me that the minor bump would be app-only but a patch bump would roll nodes, but if you can't include the flatcar update this time please just add an exception with the reason

**Comment by marians** [comment]
The version number is a UX problem. Given the discussions we have with regard to trust in upgrades, I would not recommend publishing this.

Customers understanding our versioning scheme is an important foundation for trust. This exception here makes understanding our scheme hard.

**Comment by tomahawk28** [comment]
Failed to deploy in ginger. 
```
opsctl deploy releases@helm3-aws-12-2-0 -i ginger --jumphost-user jhbok
Requesting the creation of deployment using "App" method

    chart           releases-aws
    repository      releases
    ref             helm3-aws-12-2-0
    version         0.0.0-e901dbb52d118c4d39717a1c5810e377fa9b1d3f
    installation    ginger

Requested the creation of deployment using "App" method

Waiting for deployment
Failed.
{"caller":"github.com/giantswarm/micrologger@v0.3.1/activation_logger.go:93","level":"error","stack":{"annotation":"execution failed error: deployment failed (status: `failed`, reason: `Upgrade \"releases-aws-unique\" failed: cannot patch \"v11.3.1\" with kind Release: Release.release.giantswarm.io \"v11.3.1\" is invalid: status.inUse: Required value && cannot patch \"v11.3.3\" with kind Release: Release.release.giantswarm.io \"v11.3.3\" is invalid: status.inUse: Required value && cannot patch \"v11.4.0\" with kind Release: Release.release.giantswarm.io \"v11.4.0\" is invalid: status.inUse: Required value && cannot patch \"v11.4.1\" with kind Release: Release.release.giantswarm.io \"v11.4.1\" is invalid: status.inUse: Required value && cannot patch \"v11.5.0\" with kind Release: Release.release.giantswarm.io \"v11.5.0\" is invalid: status.inUse: Required value && cannot patch \"v11.5.1\" with kind Release: Release.release.giantswarm.io \"v11.5.1\" is invalid: status.inUse: Required value && cannot patch \"v11.5.2\" with kind Release: Release.release.giantswarm.io \"v11.5.2\" is invalid: status.inUse: Required value && cannot patch \"v11.5.4\" with kind Release: Release.release.giantswarm.io \"v11.5.4\" is invalid: status.inUse: Required value && cannot patch \"v12.0.0\" with kind Release: Release.release.giantswarm.io \"v12.0.0\" is invalid: status.inUse: Required value && cannot patch \"v12.1.0\" with kind Release: Release.release.giantswarm.io \"v12.1.0\" is invalid: status.inUse: Required value && cannot patch \"v12.1.4\" with kind Release: Release.release.giantswarm.io \"v12.1.4\" is invalid: status.inUse: Required value && cannot patch \"v8.2.0\" with kind Release: Release.release.giantswarm.io \"v8.2.0\" is invalid: status.inUse: Required value && cannot patch \"v9.0.10\" with kind Release: Release.release.giantswarm.io \"v9.0.10\" is invalid: status.inUse: Required value && cannot patch \"v9.0.6\" with kind Release: Release.release.giantswarm.io \"v9.0.6\" is invalid: status.inUse: Required value && cannot patch \"v9.0.7\" with kind Release: Release.release.giantswarm.io \"v9.0.7\" is invalid: status.inUse: Required value && cannot patch \"v9.0.8\" with kind Release: Release.release.giantswarm.io \"v9.0.8\" is invalid: status.inUse: Required value && cannot patch \"v9.0.9\" with kind Release: Release.release.giantswarm.io \"v9.0.9\" is invalid: status.inUse: Required value && cannot patch \"v9.3.2\" with kind Release: Release.release.giantswarm.io \"v9.3.2\" is invalid: status.inUse: Required value && cannot patch \"v9.3.3\" with kind Release: Release.release.giantswarm.io \"v9.3.3\" is invalid: status.inUse: Required value && cannot patch \"v9.3.6\" with kind Release: Release.release.giantswarm.io \"v9.3.6\" is invalid: status.inUse: Required value && cannot patch \"v9.3.7\" with kind Release: Release.release.giantswarm.io \"v9.3.7\" is invalid: status.inUse: Required value && cannot patch \"v9.3.8\" with kind Release: Release.release.giantswarm.io \"v9.3.8\" is invalid: status.inUse: Required value`)","kind":"unknown","stack":[{"file":"/Users/jihyukbok/go/src/github.com/giantswarm/opsctl/pkg/cmd/deploy/appdeploy/deployer.go","line":309},{"file":"/Users/jihyukbok/.gvm/pkgsets/go1.13/global/pkg/mod/github.com/giantswarm/backoff@v0.2.0/retry.go","line":13},{"file":"/Users/jihyukbok/go/src/github.com/giantswarm/opsctl/pkg/cmd/deploy/appdeploy/deployer.go","line":318},{"file":"/Users/jihyukbok/go/src/github.com/giantswarm/opsctl/command/deploy/command.go","line":346},{"file":"/Users/jihyukbok/go/src/github.com/giantswarm/opsctl/pkg/output/funcs.go","line":33},{"file":"/Users/jihyukbok/go/src/github.com/giantswarm/opsctl/command/deploy/command.go","line":356}]},"time":"2020-09-16T13:32:38.527629+00:00","verbosity":0}
```
I'm not sure what's wrong. Can anybody help?

**Comment by cokiengchiara** [comment]
> The version number is a UX problem. Given the discussions we have with regard to trust in upgrades, I would not recommend publishing this.
> 
> Customers understanding our versioning scheme is an important foundation for trust. This exception here makes understanding our scheme hard.

This was resolved between @rossf7 and @marians in Slack. Summary:
* We want to use `v12.1.4-2020091600` and not `v12.1.5` specifically b/c the unusual version makes it clearer that it is a preview release, and not an actual release. 
* The release will be available only in anteater and not in other installations. Marian is fine with this from UX perspective.

**Comment by cokiengchiara** [comment]
@tomahawk28 @rossf7 since customer is already on preview release, what should be done with this PR...? can we safely close it?

**Comment by tomahawk28** [comment]
Okay, I'm going to close this.

**Comment by tomahawk28** [comment]
Still need one more patch deploying to the `anteater`.

File: aws/v12.1.4-2020091600/release.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+apiVersion: release.giantswarm.io/v1alpha1
+kind: Release
+metadata:
+  annotations:
+    helm.sh/resource-policy: keep
```
Comment (rossf7) [inline aws/v12.1.4-2020091600/release.yaml:5]: Nice 👍

**Review by rossf7 — APPROVED** [review]
LGTM

**Comment by tomahawk28** [comment]
Going to release this after standup.

**Comment by tomahawk28** [comment]
I'm closing this since 12.3.0 is released today.

**Pull request closed without merging**